src/item_location.cpp

```diff
@@ -478,6 +478,101 @@ class item_location::impl::item_on_vehicle : public item_location::impl
         }
 };
 
+class item_location::impl::item_in_container : public item_location::impl
+{
+    private:
+        item_location container;
+
+        // figures out the index for the item, which is where it is in the total list of contents
+        // note: could be a better way of handling this?
+        int calc_index() const {
+            int idx = 0;
+            for( const item &it : container->contents.all_items() ) {
+                if( target() == &it ) {
+                    return idx;
+                }
+                idx++;
+            }
+            if( idx == 0 ) {
+                return -1;
+            }
+            return idx;
+        }
+    public:
+        item_in_container( const item_location &container, item *which ) :
+            impl( which ), container( container ) {}
+
+        void serialize( JsonOut &js ) const override {
+            js.start_object();
+            js.member( "idx", calc_index() );
+            js.member( "type", "in_container" );
+            js.member( "parent", container );
+            js.end_object();
+        }
+
+        item *unpack( int idx ) const override {
+            std::list<item> all_items = container->contents.all_items();
+            auto iter = all_items.begin();
+            std::advance( iter, idx );
+            if( iter != all_items.end() ) {
+                return &*iter;
+            } else {
+                return nullptr;
+            }
+        }
+
+        std::string describe( const Character * ) const override {
+            if( !target() ) {
+                return std::string();
+            }
+            return _( string_format( "inside %s", container->tname() ) );
```

Comment:
Same translation problem as in the other PR.
```suggestion
            return string_format( _( "inside %s" ), container->tname() );
```